install/installer/pkg/components/components-ide/components.go

```diff
@@ -19,6 +22,13 @@ var Objects = common.CompositeRenderFunc(
 	ide_service.Objects,
 	ide_proxy.Objects,
 	openvsxproxy.Objects,
+	func(cfg *common.RenderContext) ([]runtime.Object, error) {
+		if cfg.Config.Kind == config.InstallationFull {
+			// ws-proxy is already included by WorkspaceComponents in a full installation.
+			return nil, nil
+		}
+		return wsproxy.Objects(cfg)
```

Comment:
this will start rolling out `ws-proxy` in application clusters on IDE installs (as eventually, `ws-proxy` is only used by IDE components to proxy to e.g. supervisor). Any concerns around this? 

Not sure where else we could put `ws-proxy` in the install once we no longer include workspace components in application clusters.

Reply:
@akosyakov and @iQQBot for :eyes:

Reply:
It is hard to say for me without understanding this PR. If an idea to move image build within the workspace cluster then  ws-proxy should not be necessary in the application cluster?

Generally IDE components are things which are installed within the workspace for end users. Things which are installed on workspace node/cluster should be under workspace components.

Reply:
From testing we saw that `ws-proxy` is still used in the application cluster for IDE proxy routes (e.g. [this supervisor route](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/frontend/src/ide/supervisor-service-client.ts#L43)). [This Slack thread](https://gitpod.slack.com/archives/C02F19UUW6S/p1670229682279999?thread_ts=1670219880.198639&cid=C02F19UUW6S) has some more context as well.

Do you think we can move those routes to workspace clusters as well? Or would that require some refactoring of supervisor?